Sources/Quick/QuickSpec.m

```diff
@@ -115,10 +115,23 @@ + (SEL)addInstanceMethodForExample:(Example *)example {
     }
 
     const char *types = [[NSString stringWithFormat:@"%s%s%s", @encode(id), @encode(id), @encode(SEL)] UTF8String];
-    NSString *selectorName = [NSString stringWithFormat:@"%@_%@_%ld",
-                              example.name.qck_selectorName,
-                              sanitizedFileName,
-                              (long)example.callsite.line];
+    
+    NSString *originalName = example.name.qck_selectorName;
+    NSString *selectorName = originalName;
+    unsigned int i = 2;
+    
+    static NSMutableSet<NSString *> *selectorNames;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        selectorNames = [[NSMutableSet alloc] init];
+    });
+    
+    while ([selectorNames containsObject:selectorName]) {
+        selectorName = [NSString stringWithFormat:@"%@_%u", originalName, i++];
+    }
```

Comment:
1. The Example class doesn't presently track the selector used for the Example. I tried adding this capability, but was having trouble assigning the selector. I declared the `var` with type `Selector` in the Example class, but it doesn't come through into the QuickSpec.m file where I would have assigned it.

2. Yeah, it would be great to test the generated selector names directly. For now, I'm addressing this with an internal `+[QuickSpec allSelectors]` method that returns an immutable copy of the global selector set, and test expectations against that.

3. Using the test described above, I verified that I'm getting two selectors for a single example. It looks like this is happening because `+[QuickSpec testInvocations]` is called twice: once in `+initialize` and again in `-[XCTestSuite testSuiteForTestCaseClass:]`. I tried removing the call in `+initialize` and didn't noticed any ill-effects (the tests passed). Is it there for backwards-compatibility with an older Xcode version, or something like that?